src/components/DailyRoutinePage/DailyRoutine.js

```diff
@@ -1,35 +1,84 @@
 import "./DailyRoutine.css";
 import AddedProductCard from "./AddedProductCard.js";
 import { FaArrowLeft } from "react-icons/fa";
+import { useParams, useHistory } from "react-router-dom";
+import { getDataFromLocalStorage } from "../../utility/localStorage.js";
+import { useEffect, useState } from "react";
 
 export default function DailyRoutine() {
+  const { weekday } = useParams();
+  const history = useHistory();
+  const [allItems, setAllItems] = useState([]);
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    const fetchProducts = () => {
+      return fetch("/products.json")
+        .then((response) => response.json())
+        .then((productData) => {
+          setProducts(productData);
+        });
+    };
+    fetchProducts();
+  }, []);
+
+  useEffect(() => {
+    const routine = getDataFromLocalStorage();
+    setAllItems(routine);
+  }, [weekday]);
+
+  let i;
+  if (weekday === "Monday") {
+    i = 0;
+  } else if (weekday === "Tuesday") {
+    i = 1;
+  } else if (weekday === "Wednesday") {
+    i = 2;
+  } else if (weekday === "Thursday") {
+    i = 3;
+  } else if (weekday === "Friday") {
+    i = 4;
+  } else if (weekday === "Saturday") {
+    i = 5;
+  } else if (weekday === "Sunday") {
+    i = 6;
+  }
+
+  const weekdayArray = allItems.filter((item) => {
+    return item.days[i].isChecked;
+  });
+
   return (
-    <div className="DailyRoutine">
+    <div className="DailyRoutine" key={weekday}>
```

Comment:
Why do you need a `key` here?. Just corious :)

Reply:
The console kept giving me the error with "Each child in a list should have a unique "key" prop."
 So i silenced it, by giving it a key :D Admittedly, I have ignored it for all other lists. What's good practice here? I don't see the issue with not having a key, but I also don't like the error 🤔

Reply:
But this is not a list, is it?
maybe it was a list before, can you try removing it now and see?

Reply:
hmmmmh, no you are right, it's not a list 🤔 
i am getting the error even when the key is there. I removed it now, since it doesn't make a difference